src/main/java/com/rtbhouse/kafka/workers/impl/KafkaWorkersImpl.java

```diff
@@ -182,19 +184,17 @@ public void close() {
             logger.error("interrupted", e);
         }
 
-        // then close and clean up any pending resources
-        for (WorkerThread<K, V> workerThread : workerThreads) {
-            closeThreadResources(workerThread);
-        }
-        closeThreadResources(consumerThread);
-        closeThreadResources(punctuatorThread);
+        closeThread(consumerThread);
```

Comment:
it is not necessary:
- both ConsumerThread and PunctuationThread should be closed successfully by previous shutdown() call
- if it is not a case it means that poll() was blocked and could not be timeouted so it is unlikely that we are able to commit now
- consumer is not thread safe so it is good to call all its methods serially

Reply:
Removed